Components/Target.gd

```diff
@@ -2,5 +2,26 @@ class_name Target
 
 extends Node
 
-var closest : Object
-var in_area : Array[Object] 
+var current : Object
+var in_range : Array[Object] 
+
+#maybe check that in_range isn't empty
+#might only be needed if exited area happens more often than shooting
+func is_in_range(target : Object,caster : BaseEntity) -> bool:
+	if in_range.has(target):
+		return true
+	else:
+		return false
+
+func find_closest(caster : BaseEntity) -> BaseEntity:
+	if in_range != []:
```

Comment:
you dont need this condition, if in_range is empty the for loop will simply not be entered